.github/workflows/check.yaml

```diff
@@ -8,30 +8,37 @@ on:
       - synchronize
 permissions:
   contents: read
+  pull-requests: write
 jobs:
-  # This job checks the PR title using
-  # https://github.com/conventional-changelog/commitlint
-  # for the conventional commit format at
-  # https://www.conventionalcommits.org/en/v1.0.0/
-  # See also /.github/commitlint.config.js for more details
-  # We only need to check the PR title because it will end up being the
-  # (default) commit title when doing squash merges with Github.
-  # See
-  # https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#merge-message-for-a-squash-merge
-  # for more info. We have "Default to PR title for squash merge commits" enabled.
   commit-lint:
     timeout-minutes: 10
-    name: "Check PR title"
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
-        with:
-          fetch-depth: 0
-      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
-      - run: npm install -g @commitlint/cli@19.0.1 @commitlint/config-conventional@19.0.0
-      - name: "Check PR title"
+      - name: Check PR title
+        # This job checks the PR title using
+        # https://github.com/conventional-changelog/commitlint
+        # for the conventional commit format at
+        # https://www.conventionalcommits.org/en/v1.0.0/
+        # See also /.github/commitlint.config.js for more details
+        # We only need to check the PR title because it will end up being the
+        # (default) commit title when doing squash merges with Github.
+        # See
+        # https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#merge-message-for-a-squash-merge
+        # for more info. We have "Default to PR title for squash merge commits" enabled.
+        if: github.event.action != 'synchronize'
         # Inject as env variable to escape properly
         env:
           TITLE: ${{ github.event.pull_request.title }}
         run: |
+          npm install -g @commitlint/cli@19.0.1 @commitlint/config-conventional@19.0.0
           commitlint --config .github/commitlint.config.js --edit=<(echo "${TITLE}")
+      - name: Maybe add backport label
```

Comment:
I don't think this works properly :( didn't have time to check why that's happening. Should we use https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore instead? (like Mike suggested some time ago). 

E.g. this one https://github.com/kumahq/kuma/pull/9430#event-11946827074 - Bart had to add a label himself.

Reply:
Yeah I've noticed it not working properly. 

Another option is to use:

```
$ gh pr view 9479 --json files  --jq '.files[].path'
.github/workflows/auto-backport.yaml
.github/workflows/check.yaml
```

In any case that's a different issue than merging this inline: https://github.com/kumahq/kuma/issues/9482